extensions/sql-database-projects/src/test/projectTree.test.ts

```diff
@@ -18,30 +18,31 @@ describe('Project Tree tests', function (): void {
 	it('Should correctly order tree nodes by type, then by name', function (): void {
 		const root = os.platform() === 'win32' ? 'Z:\\' : '/';
 
-		const parent = new ProjectRootTreeItem(new Project(vscode.Uri.file(`${root}Fake.sqlproj`).fsPath));
+		const sqlprojUri = vscode.Uri.file(`${root}Fake.sqlproj`);
+		const parent = new ProjectRootTreeItem(new Project(sqlprojUri.fsPath));
 
 		let inputNodes: (FileNode | FolderNode)[] = [
-			new FileNode(vscode.Uri.file(`${root}C`), parent),
-			new FileNode(vscode.Uri.file(`${root}D`), parent),
-			new FolderNode(vscode.Uri.file(`${root}Z`), parent),
-			new FolderNode(vscode.Uri.file(`${root}X`), parent),
-			new FileNode(vscode.Uri.file(`${root}B`), parent),
-			new FileNode(vscode.Uri.file(`${root}A`), parent),
-			new FolderNode(vscode.Uri.file(`${root}W`), parent),
-			new FolderNode(vscode.Uri.file(`${root}Y`), parent)
+			new FileNode(vscode.Uri.file(`${root}C`), sqlprojUri, parent),
```

Comment:
Do FileNodes not also need the circular reference removed?

Reply:
they do, but the actual removal will happen in the next PR. This PR was adding sqlprojUri and using that for calculating paths so these tree nodes don't use the parent node to figure that out anymore.

I split this out into a separate PR because there are several places in the projectController that use the `root` of nodes, which uses the parent reference, so that's a separate mess that needs to be cleaned up when the parent references are removed.